include/lbann/callbacks/callback.hpp

```diff
@@ -32,7 +32,6 @@
 #include "lbann/trainers/trainer.hpp"
 #include "lbann/layers/layer.hpp"
 #include "lbann/models/model.hpp"
-#include "lbann/trainers/trainer.hpp"
```

Comment:
Don't remove this line. `trainer` is used in the API. Include what you use. Or forward-declare it -- not a fight I'm really interested in picking. But the fight I am willing to pick is that you shouldn't use a symbol without somehow explicitly declaring it via either by forward declaration or by `#include`-ing the interface.

Reply:
it is a duplicate of line 32.